models/comments.js

```diff
@@ -46,4 +46,20 @@ exports.addComment = async (article_id, username, body) => {
   } catch (err) {
     return Promise.reject(err);
   }
+};
+
+exports.removeComment = async (comment_id) => {
+  const query = `DELETE FROM comments WHERE comment_id = $1`;
+  try {
+    const comment = await db.query(
+      "SELECT * FROM comments WHERE comment_id = $1",
+      [comment_id],
+    );
+    if (comment.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Comment not found" });
+    }
+    await db.query(query, [comment_id]);
+  } catch (err) {
+    return Promise.reject(err);
+  }
```

Comment:
Nice 👍 You could have potentially saved yourself from doing a check-if-exists sql request by simply checking how many rows your delete request impacted. If it was 0, you could have thrown your error. res.rowCount is the property to check in the then block, I believe 🙂

That being said, what you've done works well so it's up to you 🙂